### PR TITLE
[MIG] stock_picking_customer_ref: Migration to 10.0

### PR DESCRIPTION
**Pull request by chienandalu**
Stock picking customer reference
================================

This module displays the sale reference/description in the pickings. It also
allows you to filter and search for this new field.

Usage
=====

Edit the client reference in a sale order with pickings. It will show up in
every picking associated.

cc @Tecnativa

### CODE REVIEW

File: stock_picking_customer_ref/README.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Stock picking customer reference
+================================
+
+This module displays the sale reference/description in the pickings. It also
+allows you to filter and search for this new field.
+
+Usage
+=====
+
+Edit the client reference in a sale order with pickings. It will show up in
```
Comment (pedrobaeza) [inline stock_picking_customer_ref/README.rst:15]: Make a bit better usage steps.

File: stock_picking_customer_ref/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (pedrobaeza) [inline stock_picking_customer_ref/__init__.py]: Use new header templates

File: stock_picking_customer_ref/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# Copyright 2015 Tecnativa - Pedro M. Baeza
+# Copyright 2015 AvanzOSC - Ana Juaristi
+# Copyright 2015 AvanzOSC - Mikel Arregi
+# Copyright 2015 AvanzOSC - Alfredo de la Fuente
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    'name': 'Stock Picking Customer Ref',
+    'version': '10.0.1.0.0',
+    'category': 'Warehouse Management',
```
Comment (pedrobaeza) [inline stock_picking_customer_ref/__manifest__.py]: I think the category is now simply called `Warehouse`

File: stock_picking_customer_ref/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    def write(self, vals):
+        picking_obj = self.env['stock.picking']
+        # The field "client_order_ref" of pickings, is a calculated field. It
+        # can not be put into the field dependence "sale_id".
+        for sale in self:
+            if (vals.get('client_order_ref', False) and
+                    sale.procurement_group_id):
+                cond = [('group_id', '=', sale.procurement_group_id.id)]
+                pickings = picking_obj.search(cond)
+                pickings.write({
```
Comment (pedrobaeza) [inline stock_picking_customer_ref/models/sale_order.py]: Why writing this if it's a computed field?

File: stock_picking_customer_ref/models/stock_picking.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# Copyright 2015 AvanzOSC - Alfredo de la Fuente
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.one
```
Comment (pedrobaeza) [inline stock_picking_customer_ref/models/stock_picking.py]: Make it `@api.multi`

File: stock_picking_customer_ref/models/stock_picking.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    @api.depends('group_id')
+    def _calculate_client_order_ref(self):
+        sale_obj = self.env['sale.order']
+        self.client_order_ref = ''
```
Comment (pedrobaeza) [inline stock_picking_customer_ref/models/stock_picking.py]: No need of assigning default value.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: stock_picking_customer_ref/models/sale_order.py (hunk 4 above)
Comment (chienandalu) [reply stock_picking_customer_ref/models/sale_order.py]: Yes, the problem here seems to be that the compute field in picking is only going to be triggered when the picking is created and the ref already existed. So the aproach should be refocused to either:

- Get rid of the compute field and catch client ref on the picking creation.
- Get rid of the sale methods and try to compute depending on another field (incomplete approach?)

Your opinion is welcome.

**Review by LoisRForgeFlow — APPROVED** [review]
Tested on runbot + code reveiw :+1:

File: stock_picking_customer_ref/models/sale_order.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.multi
+    def write(self, vals):
+        # The field "client_order_ref" of pickings, is a calculated field. It
```
Comment (pedrobaeza) [inline stock_picking_customer_ref/models/sale_order.py]: Why not? A computed field can be also a dependency.

File: stock_picking_customer_ref/models/sale_order.py (hunk 4 above)
Comment (pedrobaeza) [reply stock_picking_customer_ref/models/sale_order.py]: I don't see the problem. You can specify more dependencies (even from computed fields).

**Comment by chienandalu** [comment]
@pedrobaeza Right, it was unnecesary to catch write_order write() :+1:

**Review by pedrobaeza — APPROVED** [review]

**Review by luismontalba — APPROVED** [review]
Tested on runbot

**Comment by pedrobaeza** [comment]
@chienandalu squash for merging

**Comment by chienandalu** [comment]
@pedrobaeza Done

**Pull request merged**